### PR TITLE
cmake: auto-detect xdr headers, useful for >=glibc-2.26

### PR DESCRIPTION
**Pull request by junghans**
glibc-2.26 won't have xdr headers anymore, so to avoid a compile error, let's auto-detect header availability.

### CODE REVIEW

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -171,8 +171,19 @@ if(ENABLE_KSPACE)
...
+  find_package_handle_standard_args(XDR_HEADERS REQUIRED_VARS XDR_INCLUDE_DIR)
+  if(XDR_HEADERS_FOUND)
+    option(LAMMPS_XDR "include XDR compatibility files for doing particle dumps in XTC format" OFF)
+  else()
+    option(LAMMPS_XDR "include XDR compatibility files for doing particle dumps in XTC format" ON)
+  endif()
+  if(NOT XDR_HEADERS_FOUND AND NOT LAMMPS_XDR)
+    message(FATAL_ERROR "MISC package needs xdr headers! Set LAMMPS_XDR=ON to use internal headers.")
+  elseif(XDR_HEADERS_FOUND)
+    include_directories(${XDR_INCLUDE_DIR})
```
Comment (rbberger) [inline cmake/CMakeLists.txt]: what's the dropin replacement for it? libtirpc? If so, shouldn't something be linked in as well?

**Comment by rbberger** [comment]
@junghans btw, I'm running 2.26-11 on Arch Linux without any problem, so you might want to clarify. You're talking about a future release like in Fedora 28, specifically [this](https://fedoraproject.org/wiki/Changes/SunRPCRemoval)?

File: cmake/CMakeLists.txt (hunk 1 above)
Comment (akohlmey) [reply cmake/CMakeLists.txt]: LAMMPS contains compatibility software (imported from gromacs) in `xdr_compat.h` and `xdr_compat.cpp`. Environments w/o XDR support are not new, Windows being one of them.
So this should not only apply to future Linux distributions, but should also be tested on other non-XDR OS environments.

**Comment by akohlmey** [comment]
Wouldn't it be better to merge this with the handling of LAMMPS_XDR. As far as I remember, `dump xtc` is the only code using the XDR API. Another option worth considering would be to use the internal XDR code always and get rid of the interface to the OS provided C-library.

**Comment by junghans** [comment]
Yes, I found this on Fedora 28 and Gentoo (with glibc-2.26-r5). 

I mainly want CMake to check if the header is there, set `LAMMPS_XDR` to `ON` if necessary and not fail at compile time.

Checking for `libtirpc` could be a second option beside using our internal `xdr_compat.cpp`.

**Pull request closed without merging**